planner/core/stringer.go

```diff
@@ -28,10 +28,25 @@ func ToString(p Plan) string {
 	return strings.Join(strs, "->")
 }
 
+func needIncludeChildrenString(plan Plan) bool {
+	switch x := plan.(type) {
+	case *LogicalUnionAll, *PhysicalUnionAll, *LogicalPartitionUnionAll:
+		// after https://github.com/pingcap/tidb/pull/25218, the union may contain less than 2 children,
+		// but we still wants to include its child plan's information when calling `toString` on union.
+		return true
```

Comment:
would this case happen in other cases？

Reply:
You mean will `the union may contain less than 2 children` cause other bugs? I'm not sure about this, and if it does happen, we will fix it in other prs.

Reply:
oncall 3751 can't reproduce, but maybe it's caused by the same bug? @windtalker